### PR TITLE
Fix "config.context undefined"

### PR DESCRIPTION
**Pull request by Luke265**

### CODE REVIEW

File: src/components/angular2-modal/overlay/overlay.service.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export class Overlay {
     }
 
     let dialog = new DialogRef<any>(this, config.context || {});
-    dialog.inElement = !!config.context.inElement;
+    dialog.inElement = config.context ? config.context.inElement : false;
```
Comment (shlomiassaf) [inline src/components/angular2-modal/overlay/overlay.service.ts]: Can you please use `!!` in `config.context.inElement`

Thanks!

Comment (Luke265) [inline src/components/angular2-modal/overlay/overlay.service.ts]: I'm keep getting this error when using `!!`:
````
Unhandled Promise rejection: config.context is undefined ; Zone: angular ; Task: Promise.then ; Value: TypeError: config.context is undefined
Stack trace:
Overlay</Overlay.prototype.createOverlay@http://localhost:3000/2.chunk.js:1664:29
Overlay</Overlay.prototype.open/<@http://localhost:3000/2.chunk.js:1657:41
Overlay</Overlay.prototype.open@http://localhost:3000/2.chunk.js:1656:16
Modal</Modal.prototype.open@http://localhost:3000/2.chunk.js:3170:27
````

Comment (shlomiassaf) [inline src/components/angular2-modal/overlay/overlay.service.ts]: `dialog.inElement = config.context ? !!config.context.inElement : false;` throws this error?

Comment (Luke265) [inline src/components/angular2-modal/overlay/overlay.service.ts]: My bad, sorry. Fixed :)

**Comment by shlomiassaf** [comment]
@Luke265 Thank you!

Sorry to nag again but in order for the changelog to represent the progress I need the commits to
  1) be in the `angular` format.
  2) describe what was done.

For example:
**fix(overlay): don't throw when configuration has no context**

Also, squash the commits into 1 so it won't clutter the changelog and commit history.

I know it might sound or look like unnecessary but it's actually is, the changelog is super important and the commit history as well, especially when the docs are super limited.

Thanks!

**Comment by Luke265** [comment]
@shlomiassaf Can you squash the commits? I've made those commits with online git editor and I'm not sure how to squash with it.

**Comment by shlomiassaf** [comment]
@kyledeweese yes, I'm a bit swamped now but I will try to do it soon.

@Luke265, Squashing requires force push, I hope I can do that to a repo you own, we'll see :)

**Pull request merged**